From 7b6433573fb96f91717dd23b2cc67beabb9212e6 Mon Sep 17 00:00:00 2001
From: l1ttleO <l1ttleofficial@outlook.com>
Date: Mon, 11 Jul 2022 23:25:57 +0500
Subject: [PATCH] Make botplay act like a real player

---
 source/PlayState.hx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/source/PlayState.hx b/source/PlayState.hx
index 6794a89..9e779fc 100644
--- a/source/PlayState.hx
+++ b/source/PlayState.hx
@@ -2339,7 +2339,7 @@ class PlayState extends MusicBeatState
 				new FlxTextFormatMarkerPair(blackFormat, '=')
 			]);
 
-		if(ClientPrefs.scoreZoom && !miss && !cpuControlled)
+		if(ClientPrefs.scoreZoom && !miss/* && !cpuControlled*/)
 		{
 			if(scoreTxtTween != null) {
 				scoreTxtTween.cancel();
@@ -3334,7 +3334,7 @@ class PlayState extends MusicBeatState
 							goodNoteHit(daNote);
 							if (!daNote.wasGoodHit) strainKeys[daNote.noteData] += songSpeed * (holdStrainTimer > 0 ? 0.2 : 0.05);
 						}
-					} else if(daNote.strumTime <= Conductor.songPosition || (daNote.isSustainNote && daNote.canBeHit && daNote.mustPress)) {
+					} else if(!daNote.ignoreNote && !daNote.hitCausesMiss && (daNote.strumTime <= Conductor.songPosition || (daNote.isSustainNote && daNote.canBeHit && daNote.mustPress))) {
 						goodNoteHit(daNote);
 					}
 				}
@@ -3370,7 +3370,7 @@ class PlayState extends MusicBeatState
 				// Kill extremely late notes and cause misses
 				if (Conductor.songPosition > noteKillOffset + daNote.strumTime)
 				{
-					if (daNote.mustPress && !cpuControlled &&!daNote.ignoreNote && !endingSong && (daNote.tooLate || !daNote.wasGoodHit)) {
+					if (daNote.mustPress && /*!cpuControlled &&*/!daNote.ignoreNote && !endingSong && (daNote.tooLate || !daNote.wasGoodHit)) {
 						noteMiss(daNote);
 					}
 
-- 
2.34.1.windows.1

